### DISCUSSION TITLE
Mapping .xyz to atoms in SMILES

### DISCUSSION THREAD

**Discussion by rajathsalegame** [rdkit/rdkit#3800] [Q&A]
Hi everyone,

Do any of you know whether there's an easy way to map from an `.xyz` file with 3D coordinates to atoms in a given SMILES after using `MolFromxyz`? I would like to be able to associate every atom in the `.xyz` file with an atom index in a mol.

**Comment by greglandrum** [comment]
The RDKit doesn't have code to guess bond orders from geometry, so if you really don't have any idea what the atom ordering in the XYZ file is, then I think you're probably going to have to use something like xyz2mol: https://github.com/jensengroup/xyz2mol

**Reply by rajathsalegame** [reply]
Thanks!

**Comment by xiaoruiDong** [comment]
To add on, besides the `xyz2mol` method by Jensen et al., Openbabel can also perceive molecules from xyz strings.

One step further, to match atom order in the smiles to the atom order in the perceived molecules, you can use `GetSubstructMatches`. I have some examples in one of my repos (https://github.com/xiaoruiDong/RDMC/tree/main/ipython), using RDKit for such purpose. I can describe it to you more detailedly if you want.

**Reply by rajathsalegame** [reply]
@xiaoruiDong, thank you, that might be helpful--I will take a look at the notebook first then contact you with any questions I might have.
